gc/src/gc.rs

```diff
@@ -112,8 +112,7 @@ impl<T: Trace + ?Sized> GcBox<T> {
     pub unsafe fn root_inner(&self) {
         // abort if the count overflows to prevent `mem::forget` loops that could otherwise lead to
         // erroneous drops
-        self.header.roots.set(self.header.roots.get()
-                              .checked_add(1).unwrap_or_else(|| ::std::intrinsics::abort()));
+        self.header.roots.set(self.header.roots.get().checked_add(1).unwrap());
```

Comment:
I originally made this change because I already had unwinding coming from this function in my initial pass, and figured that the extra code size overhead would be negligible, but I'll remove it now.